src/guest/start.c

```diff
@@ -572,7 +648,7 @@ static void cleanup(int num, int removed_sof_tgl_snd_module)
 	cleanup_passthrough();
 	if(removed_sof_tgl_snd_module)
 		insert_sof_tgl_snd_module();
-
+	cleanup_hugepages();
```

Comment:
When setting up SRIOV, the order is to allocate hugepages first, then add the VFIOs.
Thus for cleanup, the usual safe practice is to do it in the reverse order.
i.e. we should release the VFIOs then free the hugepages.

Due to some GPU bug, this entire cleanup process is commented out as a workaround in Android 11 ADL.
Thus once allocated, it is not released/freed.

Not sure what happens here if we free the hugepages here without removing the VFIOs.
This probably needs further testing to ensure that it won't cause any adverse effect.
Otherwise, it may be safer to just comment this out totally.